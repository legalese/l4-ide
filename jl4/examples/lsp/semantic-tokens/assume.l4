§ `Assumptions`
-- An authorised person may issue a community protection notice to an
-- individual aged 16 or over, or a body, if satisfied on reasonable grounds that—
--
--   a. the conduct of the individual or body is having a detrimental effect,
--      of a persistent or continuing nature, on the quality of life of those
--      in the locality, and
--   b. the conduct is unreasonable.

ASSUME Person IS A TYPE
ASSUME `is authorised`  IS A FUNCTION FROM Person TO BOOLEAN

ASSUME Receiver IS A TYPE
ASSUME `is an individual aged 16 or over` IS A FUNCTION FROM Receiver TO BOOLEAN
ASSUME `is a body` IS A FUNCTION FROM Receiver TO BOOLEAN
