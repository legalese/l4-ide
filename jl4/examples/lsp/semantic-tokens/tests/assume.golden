Phase: Lexer
1:1-2 SemanticTokenTypes_Operator
1:3-16 SemanticTokenTypes_Variable
2:1-70 SemanticTokenTypes_Comment
3:1-83 SemanticTokenTypes_Comment
4:1-3 SemanticTokenTypes_Comment
5:1-78 SemanticTokenTypes_Comment
6:1-78 SemanticTokenTypes_Comment
7:1-29 SemanticTokenTypes_Comment
8:1-37 SemanticTokenTypes_Comment
10:1-7 SemanticTokenTypes_Keyword
10:8-14 SemanticTokenTypes_Variable
10:15-17 SemanticTokenTypes_Keyword
10:18-19 SemanticTokenTypes_Keyword
10:20-24 SemanticTokenTypes_Keyword
11:1-7 SemanticTokenTypes_Keyword
11:8-23 SemanticTokenTypes_Variable
11:25-27 SemanticTokenTypes_Keyword
11:28-29 SemanticTokenTypes_Keyword
11:30-38 SemanticTokenTypes_Keyword
11:39-43 SemanticTokenTypes_Keyword
11:44-50 SemanticTokenTypes_Variable
11:51-53 SemanticTokenTypes_Keyword
11:54-61 SemanticTokenTypes_Variable
13:1-7 SemanticTokenTypes_Keyword
13:8-16 SemanticTokenTypes_Variable
13:17-19 SemanticTokenTypes_Keyword
13:20-21 SemanticTokenTypes_Keyword
13:22-26 SemanticTokenTypes_Keyword
14:1-7 SemanticTokenTypes_Keyword
14:8-42 SemanticTokenTypes_Variable
14:43-45 SemanticTokenTypes_Keyword
14:46-47 SemanticTokenTypes_Keyword
14:48-56 SemanticTokenTypes_Keyword
14:57-61 SemanticTokenTypes_Keyword
14:62-70 SemanticTokenTypes_Variable
14:71-73 SemanticTokenTypes_Keyword
14:74-81 SemanticTokenTypes_Variable
15:1-7 SemanticTokenTypes_Keyword
15:8-19 SemanticTokenTypes_Variable
15:20-22 SemanticTokenTypes_Keyword
15:23-24 SemanticTokenTypes_Keyword
15:25-33 SemanticTokenTypes_Keyword
15:34-38 SemanticTokenTypes_Keyword
15:39-47 SemanticTokenTypes_Variable
15:48-50 SemanticTokenTypes_Keyword
15:51-58 SemanticTokenTypes_Variable

Phase: Parse
1:1-2 SemanticTokenTypes_Operator
1:3-16 SemanticTokenTypes_Interface
2:1-70 SemanticTokenTypes_Comment
3:1-83 SemanticTokenTypes_Comment
4:1-3 SemanticTokenTypes_Comment
5:1-78 SemanticTokenTypes_Comment
6:1-78 SemanticTokenTypes_Comment
7:1-29 SemanticTokenTypes_Comment
8:1-37 SemanticTokenTypes_Comment
10:1-7 SemanticTokenTypes_Keyword
10:8-14 SemanticTokenTypes_Variable
10:15-17 SemanticTokenTypes_Keyword
10:18-19 SemanticTokenTypes_Keyword
10:20-24 SemanticTokenTypes_Keyword
11:1-7 SemanticTokenTypes_Keyword
11:8-23 SemanticTokenTypes_Variable
11:25-27 SemanticTokenTypes_Keyword
11:28-29 SemanticTokenTypes_Keyword
11:30-38 SemanticTokenTypes_Keyword
11:39-43 SemanticTokenTypes_Keyword
11:44-50 SemanticTokenTypes_Type
11:51-53 SemanticTokenTypes_Keyword
11:54-61 SemanticTokenTypes_Type
13:1-7 SemanticTokenTypes_Keyword
13:8-16 SemanticTokenTypes_Variable
13:17-19 SemanticTokenTypes_Keyword
13:20-21 SemanticTokenTypes_Keyword
13:22-26 SemanticTokenTypes_Keyword
14:1-7 SemanticTokenTypes_Keyword
14:8-42 SemanticTokenTypes_Variable
14:43-45 SemanticTokenTypes_Keyword
14:46-47 SemanticTokenTypes_Keyword
14:48-56 SemanticTokenTypes_Keyword
14:57-61 SemanticTokenTypes_Keyword
14:62-70 SemanticTokenTypes_Type
14:71-73 SemanticTokenTypes_Keyword
14:74-81 SemanticTokenTypes_Type
15:1-7 SemanticTokenTypes_Keyword
15:8-19 SemanticTokenTypes_Variable
15:20-22 SemanticTokenTypes_Keyword
15:23-24 SemanticTokenTypes_Keyword
15:25-33 SemanticTokenTypes_Keyword
15:34-38 SemanticTokenTypes_Keyword
15:39-47 SemanticTokenTypes_Type
15:48-50 SemanticTokenTypes_Keyword
15:51-58 SemanticTokenTypes_Type

Phase: Check
1:1-2 SemanticTokenTypes_Operator
1:3-16 SemanticTokenTypes_Interface
2:1-70 SemanticTokenTypes_Comment
3:1-83 SemanticTokenTypes_Comment
4:1-3 SemanticTokenTypes_Comment
5:1-78 SemanticTokenTypes_Comment
6:1-78 SemanticTokenTypes_Comment
7:1-29 SemanticTokenTypes_Comment
8:1-37 SemanticTokenTypes_Comment
10:1-7 SemanticTokenTypes_Keyword
10:8-14 SemanticTokenTypes_Type
10:15-17 SemanticTokenTypes_Keyword
10:18-19 SemanticTokenTypes_Keyword
10:20-24 SemanticTokenTypes_Keyword
11:1-7 SemanticTokenTypes_Keyword
11:8-23 SemanticTokenTypes_Variable
11:25-27 SemanticTokenTypes_Keyword
11:28-29 SemanticTokenTypes_Keyword
11:30-38 SemanticTokenTypes_Keyword
11:39-43 SemanticTokenTypes_Keyword
11:44-50 SemanticTokenTypes_Type
11:51-53 SemanticTokenTypes_Keyword
11:54-61 SemanticTokenTypes_Type
13:1-7 SemanticTokenTypes_Keyword
13:8-16 SemanticTokenTypes_Type
13:17-19 SemanticTokenTypes_Keyword
13:20-21 SemanticTokenTypes_Keyword
13:22-26 SemanticTokenTypes_Keyword
14:1-7 SemanticTokenTypes_Keyword
14:8-42 SemanticTokenTypes_Variable
14:43-45 SemanticTokenTypes_Keyword
14:46-47 SemanticTokenTypes_Keyword
14:48-56 SemanticTokenTypes_Keyword
14:57-61 SemanticTokenTypes_Keyword
14:62-70 SemanticTokenTypes_Type
14:71-73 SemanticTokenTypes_Keyword
14:74-81 SemanticTokenTypes_Type
15:1-7 SemanticTokenTypes_Keyword
15:8-19 SemanticTokenTypes_Variable
15:20-22 SemanticTokenTypes_Keyword
15:23-24 SemanticTokenTypes_Keyword
15:25-33 SemanticTokenTypes_Keyword
15:34-38 SemanticTokenTypes_Keyword
15:39-47 SemanticTokenTypes_Type
15:48-50 SemanticTokenTypes_Keyword
15:51-58 SemanticTokenTypes_Type

